### PR TITLE
Pricing in EC2-Manager

### PR DESCRIPTION
**Pull request by jhford**
I'd like to start being able to move pricing out of the aws-provisioner codebase and into the EC2 Manager codebase.

This PR accomplishes that.

### CODE REVIEW

**Comment by jhford** [comment]
This would resolve #8

File: lib/api.js [hunk 1]
```diff
@@ -512,6 +513,45 @@ api.declare({
...
+  title: 'Request prices for EC2',
+  stability: API.stability.experimental,
+  input: 'prices-request.json#',
+  output: 'prices.json#',
+  description: [
+    'Return a list of possible prices for EC2'
+  ].join(' '),
+}, async function (req, res) {
+  let prices = await this.pricing.getPrices(req.body);
+  console.dir(req.body);
```
Comment (imbstack) [inline lib/api.js]: Leftover debug?

File: lib/main.js [hunk 2]
```diff
@@ -245,6 +261,7 @@ let load = loader({
             try {
               await pollSpotRequests({ec2: ec2[region], region, state});
             } catch (err) {
+              console.dir(err.stack || err);
```
Comment (imbstack) [inline lib/main.js]: Debug?

File: lib/main.js [hunk 3]
```diff
@@ -299,13 +316,40 @@ let load = loader({
...
+        credentials: cfg.taskcluster.credentials,
+      });
+
+    },
+  },
+
+  purgequeues: {
+    requires: ['server', 'apiClient'],
+    setup: async ({server, apiClient}) => {
+      await apiClient.purgeQueues();
```
Comment (imbstack) [inline lib/main.js]: Am I reading this correctly that we're calling a method on ourselves via http? Why not do this work directly?

File: lib/pricing-poller.js [hunk 4]
```diff
@@ -0,0 +1,242 @@
...
+    return this.iterator.stop();
+  }
+
+  /**
+   * Take a list of {price, time} and return a price that we should use
+   */
+  _findSpotPriceForInstanceType(series) {
+    let now = new Date();
+    let maxPrice = 0;
+    for (let {price, time} of series) {
```
Comment (imbstack) [inline lib/pricing-poller.js]: Are you in the anti lodash camp? If not, we can replace this loop with `_.maxBy(series, 'price')` maybe?

**Comment by jhford** [comment]
So calling ourselves by http is more to implement a simple script to purge the queues, not something we're doing in production.

File: lib/pricing-poller.js (hunk 4 above)
Comment (jhford) [inline lib/pricing-poller.js]: I'd rather not use lodash

File: lib/main.js (hunk 3 above)
Comment (jhford) [inline lib/main.js]: this is just a shortcut for development

File: lib/main.js (hunk 2 above)
Comment (jhford) [inline lib/main.js]: I'm not using the debug module, but I guess using `taskcluster-lib-log` would be better here.

File: lib/api.js (hunk 1 above)
Comment (jhford) [inline lib/api.js]: yep!

**Comment by jhford** [comment]
I'm going to land this.  The purgequeues stuff is purely a convenience for development, i'd rather remove it altogether than figure out a better way to do it.

**Comment by jhford** [comment]
rebased with:
```
commit 9b747a355d8596eb54d72a9220737cd33600bfd9
Author: John Ford <john@johnford.org>
Date:   Mon Jul 10 19:26:39 2017 +0200

    Add pricing to the EC2-Manager API

commit 7789ad97cd89df6b27472bed6ba6f6c137d6ad3b
Author: John Ford <john@johnford.org>
Date:   Mon Jul 10 13:54:34 2017 +0200

    Code which fetches pricing.  Currently limited to spot-pricing

```

**Pull request closed without merging**